### PR TITLE
Corrections to make OpenID working. 

### PR DESCRIPTION
**Pull request by Bartol0**
A number of corrections to original instruction and sources (http://jasig.github.io/cas/4.0.x/protocol/OpenID-Protocol.html) have to be made for this configuration to work.

### CODE REVIEW

File: cas-server-documentation/protocol/OpenID-Protocol.md [hunk 1]
```diff
@@ -188,6 +196,26 @@ We are done with the delegates. Now we must create the Delegating controller its
 
 Don't forget to include the *util* namespace if you don't have it already!
 
+Change EL expresiion parser to OGNL:
```
Comment (leleuj) [inline cas-server-documentation/protocol/OpenID-Protocol.md]: Small typo: *expression*

**Comment by leleuj** [comment]
Thanks for taking time to report back.

I'm surprised by the work around you propose: replacing the Spring EL by OGNL. How can it be that using Spring EL makes OpenID support broken? What was the error you got?

**Comment by Bartol0** [comment]
W dniu 17.02.2015 15:07, LELEU J�r�me pisze:
> Thanks for taking time to report back.
> 
> I'm surprised by the work around you propose: replacing the Spring EL by
> OGNL. How can it be that using Spring EL makes OpenID support broken?
> What was the error you got?

See Java exception:

2015-01-23 14:00:52,537 DEBUG
[org.jasig.cas.web.FlowExecutionExceptionResolver] - <Ignoring the
received exception due to a type mismatch>
org.springframework.expression.spel.SpelParseException: EL1041E:(pos
51): After parsing a valid expression, there is still more data in the
expression: 'neq'

for such an expression:

test="externalContext.requestParameterMap['openid.mode'] neq ''

**Comment by leleuj** [comment]
OK. In that case, why don't we update the test in the flow to be Spring EL compliant?

**Comment by Bartol0** [comment]
W dniu 17.02.2015 15:17, LELEU J�r�me pisze:
> OK. In that case, why don't we update the test in the flow to be Spring
> EL compliant?

Yes, that would be more clear, but for me introducing OGNL was easier
than learning EL.

**Comment by leleuj** [comment]
I understand, but Spring EL is definitely the target.

**Comment by serac** [comment]
Changing from SpEL is definitely not an option.

**Comment by mmoayyed** [comment]
So this is just a matter of learning a new way to configure Spring, or did you actually run into an issue?

**Comment by Bartol0** [comment]
The first case. I am not familiar with Spring.

**Comment by mmoayyed** [comment]
OK, thanks for the update. I'd recommend that you document your experiences here instead:
https://wiki.jasig.org/display/CAS/Home

**Comment by leleuj** [comment]
Just to provide an update on this. I tested a CAS server v4.0.x with OpenID support: https://github.com/leleuj/cas-openid-demo/commit/743d17eddef8f64099953477d2d07643019bd218.

For the OGNL / SpringEL issue, it's fairly easy: the `neq` OGNL keyword is to be replaced by the `ne` SpringEL keyword!

**Comment by mmoayyed** [comment]
Cool, it sounds like we can update the docs with this new info then?

**Comment by leleuj** [comment]
Yes, I'll update the doc with the code.

**Pull request closed without merging**